astropy_timeseries/downsample.py

```diff
@@ -12,11 +12,16 @@
 
 def reduceat(array, indices, function):
     """
-    Manual reduceat functionality for cases where Numpy functions don't have a reduceat
+    Manual reduceat functionality for cases where Numpy functions don't have a reduceat.
+    It will check if the input function has a reduceat and call that if it does.
     """
-    result = [function(array[indices[i]:indices[i+1]]) for i in range(len(indices) - 1)]
-    result.append(function(array[indices[-1]:]))
-    return np.array(result)
+    # Taken some inspiration from TableGroups.aggregate.
+    if hasattr(function, 'reduceat'):
+        return np.array(function.reduceat(array, indices))
+    else:
+        result = [function(array[indices[i]:indices[i+1]]) for i in range(len(indices) - 1)]
```

Comment:
I think here we should expand the loop and when indices[i+1] < indices[i] we should add a -1 in the slice

Reply:
(that will fix the discrepancy with numpy's reduceat)